pkg/telemetry/setup.go

```diff
@@ -0,0 +1,96 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+)
+
+// Global internal tracer that generates spans.
+var tracer = otel.Tracer("undefined")
+
+// A simple helper that configures OpenTelemetry for the SFU.
+func SetupTelemetry(config Config) (*tracesdk.TracerProvider, error) {
+	// Create a new resource.
+	res, err := NewResource(config.Package, config.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create a new Jaeger exporter.
+	exp, err := NewJaegerExporter(config.JaegerURL)
```

Comment:
Interesting. I've totally forgotten about the OTLP protocol. I've just checked how to add the support of OTLP in Go and noticed that there are 2 ways to send things over OTLP:

- HTTP with protobuf encoded content
- gRPC

I've checked the Jaeger setup from the branch you shared with me and if I get it right, our logger instance supports both. Does it mean I can choose any of these (depending on which one is simpler to set up) for the SFU? Do you know if I need to attach any credentials to it? I've noticed that there is a possibility to pass the token:

```go
      otlptracegrpc.WithHeaders(map[string]string{
          "Authorization": "<ADD YOUR TOKEN HERE>",
      }))
```

Reply:
The docker setup is the jaeger all-in-one which does support a wide set of protocols, but if we're able to use OTLP over HTTP then this will be consistent with element call. I don't think we'll need any access tokens.

Reply:
Makes sense! Added support of OTLP (the option to use Jaeger remains, the user can choose the exporter by configuring one or another one, i.e. I use the jaeger URL locally for the tests, but I've just tested it with the OTLP exporter and it also works fine as long as the local instance of Jaeger configures OTLP).